src/actions/measurements.ts

```diff
@@ -203,63 +264,92 @@ const parseMeasurementsJSON = (json) => {
      * Then loop over measurements to add values
      * If there are no JSON defined filters, then add all fields as filters
      */
-    const collectionFiltersArray = collection.filters;
+    const collectionFiltersArray = jsonCollection.filters;
     collection.filters = new Map(
-      (collection.filters || [])
+      (jsonCollection.filters || [])
         .map((filterField) => [filterField, {values: new Set()}])
     );
 
     // Create a temp object for groupings to keep track of values and their
     // counts so that we can create a stable default order for grouping field values
-    const groupingsValues = collection.groupings.reduce((tempObject, {key}) => {
-      tempObject[key] = {};
+    const groupingsValues: GroupingValues = jsonCollection.groupings.reduce((tempObject, {key}) => {
+      tempObject[key] = new Map();
       return tempObject;
     }, {});
 
-    collection.measurements.forEach((measurement, index) => {
-      Object.entries(measurement).forEach(([field, fieldValue]) => {
-        // Add remaining field titles
-        if (!collection.fields.has(field)) {
-          collection.fields.set(field, {title: field});
-        }
+    collection.measurements = jsonCollection.measurements.map((jsonMeasurement, index) => {
```

Comment:
suggestion/nit: a return type on the `map` would be good. (and for all the `map`s nested in this `map`…)

Reply:
Added in https://github.com/nextstrain/auspice/pull/1910/commits/4aec415702f6761110285ed0bce8cab41cc4cd13